blockchain/storage/leveldb/leveldb_test.go

```diff
@@ -37,3 +48,227 @@ func newStorage(t *testing.T) (storage.Storage, func()) {
+			Root:              types.StringToHash("mockhashstring"),
+			TransactionType:   types.LegacyTx,
+			GasUsed:           uint64(100000),
+			Status:            &status,
+			Logs:              logs,
+			CumulativeGasUsed: uint64(100000),
+			ContractAddress:   &types.Address{0xaa, 0xbb, 0xcc, 0xdd, 0xab, 0xac},
+		}
+	}
+
+	for i := 0; i < 5; i++ {
+		b.Receipts[i].LogsBloom = types.CreateBloom(b.Receipts)
+	}
+
+	return b
+}
+
+func newStorageP(t *testing.T) (storage.Storage, func(), string) {
+	t.Helper()
+
+	p, err := os.MkdirTemp("", "leveldbtest")
+	require.NoError(t, err)
+
+	require.NoError(t, os.MkdirAll(p, 0755))
+
+	s, err := NewLevelDBStorage(p, hclog.NewNullLogger())
+	require.NoError(t, err)
+
+	closeFn := func() {
+		require.NoError(t, s.Close())
+
+		if err := s.Close(); err != nil {
+			t.Fatal(err)
+		}
+
+		require.NoError(t, os.RemoveAll(p))
+	}
+
+	return s, closeFn, p
+}
+
+func countLdbFilesInPath(path string) int {
+	pattern := filepath.Join(path, "*.ldb")
+
+	files, err := filepath.Glob(pattern)
+	if err != nil {
+		return -1
+	}
+
+	return len(files)
+}
+
+func generateBlocks(t *testing.T, count int, ch chan *types.FullBlock, ctx context.Context) {
+	t.Helper()
+
+	ticker := time.NewTicker(time.Second)
+
+	for i := 1; i <= count; i++ {
+		b := generateBlock(t, uint64(i))
+		select {
+		case <-ctx.Done():
+			close(ch)
+			ticker.Stop()
+
+			return
+		case <-ticker.C:
+			ch <- b
+		}
+	}
+}
+
+func DirSize(path string) (int64, error) {
+	var size int64
+
+	err := filepath.Walk(path, func(_ string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if !info.IsDir() {
+			size += info.Size()
+		}
+
+		return err
+	})
+
+	return size, err
+}
+
+func TestWriteFullBlock(t *testing.T) {
+	t.Helper()
+
+	s, _, path := newStorageP(t)
+	defer s.Close()
+
+	count := 100
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute*45)
+
+	signchan := make(chan os.Signal, 1)
+	signal.Notify(signchan, syscall.SIGINT)
+
+	go func() {
+		<-signchan
+		cancel()
+	}()
+
+	blockchain := make(chan *types.FullBlock, 1)
+	go generateBlocks(t, count, blockchain, ctx)
+
+insertloop:
+	for i := 1; i <= count; i++ {
+		select {
+		case <-ctx.Done():
+			break insertloop
+		case b := <-blockchain:
+			batch := s.NewBatch()
+			batchHelper := storage.NewBatchHelper(batch)
+
+			batchHelper.WriteBody(b.Block.Hash(), b.Block.Body())
+
+			for _, tx := range b.Block.Transactions {
+				batchHelper.WriteTxLookup(tx.Hash, b.Block.Hash())
+			}
+
+			batchHelper.WriteHeader(b.Block.Header)
+			batchHelper.WriteHeadNumber(uint64(i))
+			batchHelper.WriteHeadHash(b.Block.Header.Hash)
+			batchHelper.WriteReceipts(b.Block.Hash(), b.Receipts)
+
+			if err := batch.Write(); err != nil {
+				fmt.Println(err)
+				t.FailNow()
+			}
+
+			fmt.Println("writing block", i)
+
+			size, _ := DirSize(path)
+			fmt.Println("\tldb file count:", countLdbFilesInPath(path))
+			fmt.Println("\tdir size", size/1_000_000, "MBs")
```

Comment:
Probably it is better to rely on `t.Log` (or `t.Logf`) if we are about to keep these outputs.